src/prefect/tasks/kubernetes/secrets.py

```diff
@@ -0,0 +1,125 @@
+import base64
+
+from typing import Any, cast, Callable, Optional
+
+from kubernetes import client
+
+from prefect.tasks.secrets.base import SecretBase
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.utilities.kubernetes import get_kubernetes_client
+
+
+class KubernetesSecret(SecretBase):
+    """
+    Task for creating a Prefect secret from the kubernetes secret.
+
+    This task will read the secret object from kubernetes and extract the value of the secret_key
+    and decode it. The kubernetes secret can be created by sealedsecret or vault and read here.
+
+    Note that you need to ensure that you have the right RBAC from your cluster admin to read the secret
+    Note that all initialization arguments can optionally be provided or overwritten at runtime.
+
+    1. Attempt to use a Prefect Secret that contains a Kubernetes API Key. If
+    `kubernetes_api_key_secret` = `None` then it will attempt the next two connection
+    methods. By default the value is `KUBERNETES_API_KEY` so providing `None` acts as
+    an override for the remote connection.
+    2. Attempt in-cluster connection (will only work when running on a Pod in a cluster)
+    3. Attempt out-of-cluster connection using the default location for a kube config file
+
+
+    Args:
+        - secret_name (string, optional): The name of the kubernetes secret object
+        - secret_key (string, optional): The key to look for in the kubernetes data
```

Comment:
What do you think about making this optional, in which case a mapping of `{key: decoded_value}` is returned for all items in the k8s secret?

Reply:
Could make it optional. However, I'm not sure of the value. In this case, we would return a Dict[str. str]. but when I think of secret I think of it as one key. The secret_name map to the k8s secret (the store) and the secret key return the value we want to extract.
If we allow it to be optional we would return for example
{"PASSWORD_DB":  "132456", "PASSWORD_S3": "654321"} 
we would have to map them later on. In this case, we can be strict about ensuring the key exists 

What do you think ? I prefer been strict here

Reply:
The use case I was thinking of was some prefect secrets expect JSON values - using a k8s secret with with multiple keys feels like a natural way to write this, rather than doing nesting below a single key. Leaving this out for now is fine though, can always add it later if users want it.